test/java/src/com/ibm/streamsx/topology/test/api/IsolateTest.java

```diff
@@ -21,54 +23,87 @@
-        
-        Condition<List<String>> condss1 = tester.stringContents(ss1, "");      
+
+        Condition<List<String>> condss1 = tester.stringContents(ss1, "");
         Condition<List<String>> condss2 = tester.stringContents(ss2, "");
-        
-	// Jenkins seems to fail when running this with a distributed tester.
-	// Not sure why, but the tests pass on my local build. 
-        /*try{
-        StreamsContextFactory
-                .getStreamsContext(StreamsContext.Type.DISTRIBUTED_TESTER)
-                .submit(topology).get(90, TimeUnit.SECONDS);
-        } catch(Exception e){
+
+        // Jenkins seems to fail when running this with a distributed tester.
+        // Not sure why, but the tests pass on my local build.
+        try {
+            StreamsContextFactory
+                    .getStreamsContext(StreamsContext.Type.DISTRIBUTED_TESTER)
+                    .submit(topology).get(90, TimeUnit.SECONDS);
+        } catch (Exception e) {
             e.printStackTrace();
         }
         Integer result1 = Integer.parseInt(condss1.getResult().get(0));
         Integer result2 = Integer.parseInt(condss2.getResult().get(0));
-        
+
         Set<Integer> m = new HashSet<>();
 
         m.add(result1);
         m.add(result2);
-        assertTrue(m.size() == 2);*/
-	StreamsContextFactory
-	    .getStreamsContext(StreamsContext.Type.TOOLKIT)
-	    .submit(topology).get();
+        assertTrue(m.size() == 2);
+    }
+    
+    @Test
+    public void isolateIsEndOfStreamTest() throws Exception {
+        assumeTrue(SC_OK);
+        Topology topology = new Topology("isolationTest");
+
+        // Construct topology
+        TStream<String> ss = topology.strings("hello");
+        TStream<String> ss1 = topology.strings("hello");
+        TStream<String> un = ss.union(ss1);
+        un.isolate();
+        StreamsContextFactory.getStreamsContext(StreamsContext.Type.TOOLKIT)
+                .submit(topology).get();
+    }
+
+    @Test
+    public void multipleIsolationTest() throws Exception {
+        Topology topology = new Topology("isolationTest");
+
+        TStream<String> ss = topology.strings("hello", "world");
+        TStream<String> ss0 = ss.isolate();
+        TStream<String> ss1 = ss0.transform(getPEId(), String.class);
+        ss1.isolate().transform(getPEId(), String.class)
+                .transform(getPEId(), String.class).print();
+
+        TStream<String> ss3 = ss.transform(getPEId(), String.class).isolate();
+        TStream<String> ss4 = ss3.transform(getPEId(), String.class).isolate();
+        TStream<String> ss5 = ss4.transform(getPEId(), String.class).isolate();
+        ss5.transform(getPEId(), String.class).print();
+
+        TStream<String> ss7 = ss3.transform(getPEId(), String.class);
+
+        StreamsContextFactory.getStreamsContext(StreamsContext.Type.TOOLKIT)
+                .submit(topology).get();
     }
 
     @Test(expected = IllegalStateException.class)
```

Comment:
Some comments as to why this should throw an exception, it looks valid to me, or is it a case that is just not yet covered?